app/src/components/Modals/add-activity-modal.tsx

```diff
@@ -333,10 +361,11 @@ const AddActivityModal: FC<AddUserModalProps> = ({
                       {...register("activity.emissionFactorType")}
```

Comment:
I think you need to change the keys that are being written to depending on the `name` property of the formInput field?